File: merkle/proof.py
from common import util

class Proof(object):
	def __init__(self, proof_type: bool=True, index: str='', proof_id: util.bitarray=None, copath: list=None) -> None:
		super(Proof, self).__init__()
		
		self.proof_type = proof_type
		self.index = index 
		self.proof_id = proof_id
		self.copath = []
		if copath != None:
			self.copath = copath

File: go/merkle/proof.go
package merkle

import "math/big"
import "fmt"

type ProofType bool

const (
	NONMEMBERSHIP    ProofType = false
	MEMBERSHIP       ProofType = true
)

type CoPathPair struct {
	ID        string
	digest    []byte
}

type Proof struct {
	proofType    ProofType
	queryID      string
	proofID      string
	coPath       []CoPathNode
}

func MakeProof(proofType ProofType, queryID string, proofID big.Int, coPath []CoPathNode) (*Proof) {
	return &Proof{proofType, queryID, proofID, coPath}
}
